torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -2626,6 +2649,122 @@ def test_no_faulty_messages(self):
     def test_custom_messages_to_delay(self):
         self.assertEqual(self.rpc_backend_options.messages_to_delay, {"SCRIPT_CALL": 1.5})
 
+    @dist_init(faulty_messages=["PYTHON_REMOTE_CALL"])
+    def test_remote_message_dropped_pickle(self):
+        if self.rank != 0:
+            return
+        dst_rank = (self.rank + 1) % self.world_size
+        dst_worker = "worker{}".format(dst_rank)
+        # Since we fail python_remote_call messages synchronously, the future
+        # corresponding to this remote call will be marked with an error when
+        # this function returns.
+        rref = rpc.remote(dst_worker, my_sleep_func, args=(1, ))
+        # Call to ensure pending callbacks are run.
+        wait_until_pending_users_flushed()
+        # Attempt to fork the RRef should raise an error indicating the rpc.remote timeout.
+        with self.assertRaisesRegex(RuntimeError, "RRef creation"):
+            rref._serialize()
+        # Test that using RRef as arg over RPC (which forks) results in the same
+        # error
+        with self.assertRaisesRegex(RuntimeError, "RRef creation"):
+            rpc.rpc_async(dst_worker, add_rref_to_value, args=(rref, 1))
+
+    def _test_remote_message_dropped_timeout(self, func, args):
+        if self.rank != 0:
+            return
+        # test the case where rpc.remote() message creation is completely dropped.
+        dst_rank = (self.rank + 1) % self.world_size
```

Comment:
Let's also test for sending to self.

Reply:
Good catch! The next version of this diff will have thes updates.